src/sys/unix.rs

```diff
@@ -535,9 +535,13 @@ pub fn preferred_languages(arena: &Arena) -> Vec<ArenaString<'_>, &Arena> {
 
     for key in ["LANGUAGE", "LC_ALL", "LANG"] {
         if let Ok(val) = std::env::var(key) {
-            locales.extend(
-                val.split(':').filter(|s| !s.is_empty()).map(|s| ArenaString::from_str(arena, s)),
-            );
+            locales.extend(val.split(':').filter(|s| !s.is_empty()).map(|s| {
+                // Replace all underscores with dashes,
+                // because the localization code expects pt-br, not pt_BR.
+                let mut res = Vec::new_in(arena);
+                res.extend(s.as_bytes().iter().map(|&b| if b == b'_' { b'-' } else { b }));
+                unsafe { ArenaString::from_utf8_unchecked(res) }
```

Comment:
I broke this by giving wrong direction over in #104. This code has the benefit of being more compact (no extra string copies).
(Iterators in Rust have size hints so the .extend() call knows exactly how many items it needs to expect and can preallocate accordingly.)